src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -71,5 +71,23 @@ public IActionResult GetInstanceInfo()
         {
             return Ok(_instanceManager.GetInstanceInfo());
         }
+
+        [HttpPost]
+        [Route("admin/instance/disable")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevel)]
+        public async Task<IActionResult> Disable([FromServices] IScriptHostManager hostManager)
+        {
+            if (_environment.IsLinuxContainerEnvironment())
+            {
+                _logger.LogInformation("Disabling container");
+                // Mark the container disabled. We check for this on host restart
+                await Utility.MarkContainerDisabled();
+                var unused = hostManager.RestartHostAsync();
+                return Ok();
+            }
+
+            _logger.LogWarning("Disable container used in non-linux environment");
```

Comment:
nit: "Disable can only be used in Linux Container environments"

Reply:
Fixed.